pyveda/vedaset/store/vedabase.py

```diff
@@ -214,10 +212,25 @@ def __del__(self):
 
 
 class VedaBase(H5DataBase):
-
     class _MetaSample(tables.IsDescription):
         vid = tables.StringCol(36)
 
+    def __init__(self, *args, dataset_id=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        if dataset_id is not None:
+            self._root._v_attrs["dataset_id"] = dataset_id
+
+    def __iter__(self):
+        img_iter = self._root.images.iterrows()
+        lbl_iter = self._root.labels.iterrows()
+        return(zip(img_iter, lbl_iter))
+
+    def __next__(self):
+       try:
+            return super().__next__()
+       except StopIteration as si:
```

Comment:
Why catch and reraise here?  Raising StopIteration is the intended functionality in defining `__next__`. Other python  object that consume iterators often depend on this error in defining their behavior.